webapp/src/index.tsx

```diff
@@ -30,19 +30,21 @@ export default class Plugin {
                     return false;
                 }
 
-                // console.debug('reply_count=', post.reply_count, '  rootdel=', post.props?.rootdel, '  post_id=', post.id); //eslint-disable-line no-console
+                // console.debug('reply_count=', post.reply_count, '  rootdel=', post.props?.rootdel, '  root_id=', post.root_id, '  post_id=', post.id); //eslint-disable-line no-console
 
-                // check if post has replies
-                if (post.reply_count === 0) {
+                // hide menu if this is not a root post (non-empty root_id means this is a reply post) or has no replies.
+                if (post.root_id || post.reply_count === 0) {
                     return false;
                 }
 
-                // check if post is already marked as root post deleted
+                // hide menu if post is already marked as root post deleted
                 if (post.props && post.props.rootdel) {
                     return false;
                 }
 
-                // Check if the user has permissions to edit his own post or edit other's posts if not the author
+                // Check if the user has permissions to edit his own post, or other's posts if not the author.
+                // We check for edit permissions instead of delete since on the back-end we will be editing the
+                // post and making it immutable.
```

Comment:
Will there be a separate PR to add `delete` in addition to edit permissions?

Reply:
I thought about it but can't find a reason to check delete permissions. Even removing attachments is considered a post edit in the other plugin.  The only permissions checked on the backend are for edit.